tests/suite/upstream_settings_test.go

```diff
@@ -0,0 +1,520 @@
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			DescribeTable("are set properly for",
+				func(expCfgs []framework.ExpectedNginxField) {
+					for _, expCfg := range expCfgs {
+						Expect(framework.ValidateNginxFieldExists(conf, expCfg)).To(Succeed())
+					}
+				},
+				Entry("Coffee upstream", []framework.ExpectedNginxField{
+					{
+						Directive: "zone",
+						Value:     "uspolicy_coffee_80 512k",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive",
+						Value:     "100",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive_requests",
+						Value:     "55",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive_time",
+						Value:     "1m",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+					{
+						Directive: "keepalive_timeout",
+						Value:     "5h",
+						Upstream:  "uspolicy_coffee_80",
+						File:      "http.conf",
+					},
+				}),
+				Entry("Tea upstream", []framework.ExpectedNginxField{
+					{
+						Directive: "zone",
+						Value:     "uspolicy_tea_80 128k",
+						Upstream:  "uspolicy_tea_80",
+						File:      "http.conf",
+					},
+				}),
+			)
+		})
+	})
+
+	When("UpstreamSettingsPolicy targets a Service that does not exists", func() {
+		Specify("upstreamSettingsPolicy sets no condition", func() {
+			files := []string{"upstream-settings-policy/invalid-svc-usps.yaml"}
+
+			Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+
+			nsname := types.NamespacedName{Name: "usps-target-not-found", Namespace: namespace}
+			Consistently(
+				func() bool {
+					return waitForUSPolicyToHaveAncestor(
+						nsname,
+					) != nil
+				}).WithTimeout(timeoutConfig.GetTimeout).
+				WithPolling(500 * time.Millisecond).
+				Should(BeTrue())
+
+			Expect(resourceManager.DeleteFromFiles(files, namespace)).To(Succeed())
+		})
+	})
+
+	When("UpstreamSettingsPolicy targets a Service that is owned by an invalid Gateway", func() {
+		Specify("upstreamSettingsPolicy has no condition set", func() {
+			// delete existing gateway
+			gatewayFileName := "upstream-settings-policy/gateway.yaml"
+			Expect(resourceManager.DeleteFromFiles([]string{gatewayFileName}, namespace)).To(Succeed())
+
+			files := []string{"upstream-settings-policy/invalid-target-usps.yaml"}
+			Expect(resourceManager.ApplyFromFiles(files, namespace)).To(Succeed())
+
+			nsname := types.NamespacedName{Name: "soda-svc-usp", Namespace: namespace}
+			gatewayName = "gateway-not-valid"
+			Expect(waitForUSPolicyStatus(
+				nsname,
+				gatewayName,
+				metav1.ConditionFalse,
+				v1alpha2.PolicyReasonTargetNotFound,
+			)).To(Succeed())
+
+			Expect(resourceManager.DeleteFromFiles(files, namespace)).To(Succeed())
+		})
+	})
+})
+
+func waitForUSPolicyToHaveAncestor(usPolicyNsName types.NamespacedName) error {
```

Comment:
This is better, but it still has a couple problems:
1. The name is misleading. We are checking that the policy does not have ancestors in its status. 
2. We are already polling using the Consistently matcher,  we don't need to also poll here. 

```
Consistently(
        func() bool {
	        return USPolicyHasNoAncestor()
        }).WithTimeout(timeoutConfig.GetTimeout).
        WithPolling(500 * time.Millisecond).
        Should(BeTrue())
```

```
func USPolicyHasNoAncestors(usPolicyNsName types.NamespacedName) bool {
   GinkgoWriter.Printf("Checking that UpstreamSettingsPolicy %q has no ancestors in status", usPolicyNsName)

   ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetStatusTimeout)
   defer cancel()

   var usPolicy ngfAPI.UpstreamSettingsPolicy
   var err error
   if err = k8sClient.Get(ctx, usPolicyNsName, &usPolicy); err != nil {
        GinkgoWriter.Printf("Failed to get UpstreamSettingsPolicy %q: %s", uspolicyNsName, err.Error())
	return false
   }
   
   return len(usPolicy.Status.Ancestors) == 0 
}
```

Reply:
I have updated this polling functionality using `wait.PollUntilContextCancelled` to be consistent with other polling functions in our tests as per @sjberman [comment](https://github.com/nginxinc/nginx-gateway-fabric/pull/2869#discussion_r1894261689)